src/Compilers/Core/Portable/Syntax/LineDirectiveMap.LineMappingEntry.cs

```diff
@@ -86,5 +86,14 @@ public int CompareTo(LineMappingEntry other)
                 return this.UnmappedLine.CompareTo(other.UnmappedLine);
             }
         }
+
+
+        protected class LineMappingEntryComparer : System.Collections.IComparer
```

Comment:
1) Is there a reason this protected?
2) Can we doc around why we needed to do this (ie Xamarin.iOS doesn't calling through IComparable instances on structs, or whatever the reason)?
3) Make this a singleton, and use it - no need to allocate on every usage here.